src/typing_extensions.py

```diff
@@ -2000,6 +2001,36 @@ def decorator(cls_or_fn):
         return decorator
 
 
+if hasattr(typing, "override"):
+    override = typing.override
+else:
+    _F = typing.TypeVar("_F", bound=typing.Callable[..., typing.Any])
+
+    def override(__arg: _F) -> _F:
+        """Indicate that a method overrides a method in a base class.
+
+        Usage:
+
+            class Base:
+                def method(self) -> None: ...
+                    pass
+
+            class Child(Base):
+                @override
+                def method(self) -> None:
+                    super().method()
+
+        When this decorator is applied to a method, the type checker will
+        validate that it overrides a method with the same name on a base class.
+        This helps prevent bugs that may occur when the base class is changed
+        without an equivalent change to a child class.
```

Comment:
Nit: I think "the base" and "a child" sounds funny to me. I'd probably us "a" in both places:
```suggestion
        This helps prevent bugs that may occur when a base class is changed
        without an equivalent change to a child class.
```

Reply:
Thanks, good point